c/src/metta.rs

```diff
@@ -335,6 +582,10 @@ pub extern "C" fn get_atom_types(space: *const space_t, atom: *const atom_ref_t,
 // MeTTa Intperpreter Interface
 // =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
 
+//QUESTION: It feels like a runner_state_t and a step_result_t are getting at the same functionality,
```

Comment:
As we discussed, I left the separate interpreter API in for now.

I added the capabilities to the RunnerState such that it could emulate the behavior of the Interpreter API with a temporary MeTTa, but it is much higher overhead to call through a RunnerState than just to call the interpreter directly.

We may revisit this if grounded atom execution is reworked to depend on a context with access to a MeTTa runner.  https://github.com/trueagi-io/hyperon-experimental/issues/410